providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/UnleashProvider.java

```diff
@@ -0,0 +1,171 @@
+package dev.openfeature.contrib.providers.unleash;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderEventDetails;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import io.getunleash.DefaultUnleash;
+import io.getunleash.Unleash;
+import io.getunleash.UnleashContext;
+import io.getunleash.Variant;
+import io.getunleash.util.UnleashConfig;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.getunleash.Variant.DISABLED_VARIANT;
+
+/**
+ * Provider implementation for Unleash.
+ */
+@Slf4j
+public class UnleashProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "Unleash Provider";
+    public static final String NOT_IMPLEMENTED =
+        "Not implemented - provider does not support this type. Only boolean is supported.";
+
+    public static final String PROVIDER_NOT_YET_INITIALIZED = "provider not yet initialized";
+    public static final String UNKNOWN_ERROR = "unknown error";
+
+    @Getter(AccessLevel.PROTECTED)
+    private UnleashOptions unleashOptions;
+
+    @Setter(AccessLevel.PROTECTED)
+    @Getter
+    private Unleash unleash;
+
+    @Setter(AccessLevel.PROTECTED)
+    @Getter
+    private ProviderState state = ProviderState.NOT_READY;
+
+    private AtomicBoolean isInitialized = new AtomicBoolean(false);
+
+    /**
+     * Constructor.
+     * @param unleashOptions UnleashOptions
+     */
+    public UnleashProvider(UnleashOptions unleashOptions) {
```

Comment:
should accept `UnleashConfig` and not a custom wrapper. This breaks the expectations of Unleash.

Reply:
The challenge is that UnleashOptions is mutable and this can lead to a race condition. If the provider needs to expend with custom options (which I really do not think is the case) it should unwrap the immutable object and create a new configuration based on the the provided object. 

There is a reason the SDK does not accept UnleashOptions in the first place.

Reply:
UnleashOptions is not Mutable.
It gets UnleashConfig.Builder, and building new UnleashConfig on initialization.
I think initialization is fine now:

```
UnleashConfig.Builder unleashConfigBuilder =
	UnleashConfig.builder()
		.unleashAPI(new URI(unleashAPI))
		.appName("fakeApp")
		.subscriber(testSubscriber)
		.synchronousFetchOnInitialisation(synchronousFetchOnInitialisation);

UnleashOptions unleashOptions = UnleashOptions.builder()
	.unleashConfigBuilder(unleashConfigBuilder)
	.build();
return new UnleashProvider(unleashOptions);
```

Reply:
Ok, I see your point. It can be confusing to maintain `UnleashConfig` and `UnleashOptions` and explain the difference. 

Maybe we can simply rename to "UnleashProviderConfig" or UnleashProviderOptions"?